### PR TITLE
filtering k3s agent config from cluster options

### PR DESCRIPTION
**Pull request by rishi-anand**
Generating k3s-agent config from cluster options 
@paynejacob, @jzhoucliqr Please review

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -18,43 +19,41 @@ const (
...
+		}
 	}
 
 	systemName := serverSystemName
 	if cluster.Role == clusterplugin.RoleWorker {
 		systemName = agentSystemName
 	}
 
-	// ensure we always have  a valid user config
-	if cluster.Options == "" {
```
Comment (paynejacob) [inline main.go:50]: this should be moved above the switch statement, this implementation makes it possible for control plane and init nodes to have invalid configuration.

File: api/agent_config.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package api
+
+type K3sAgentConfig struct {
+	Config string `json:"config,omitempty" yaml:"config,omitempty"`
```
Comment (paynejacob) [inline api/agent_config.go:4]: I don't think there is a case where a config can reference another config.

**Review by paynejacob — CHANGES_REQUESTED** [review]

File: api/agent_config.go (hunk 2 above)
Comment (jzhoucliqr) [inline api/agent_config.go:4]: Is there any properties valid for agent but not for the server?

File: main.go (hunk 1 above)
Comment (rishi-anand) [reply main.go:50]: @paynejacob If cluster.option is `yaml` then we don't need to initialize it. Earlier change of initializing cluster.option with `{}` was invalid as we were initializing it with empty json.

File: api/agent_config.go (hunk 2 above)
Comment (rishi-anand) [reply api/agent_config.go:4]: don't know @jzhoucliqr. @paynejacob may know more.

File: api/k3s_config.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package api
+
+type K3sConfig struct {
```
[the remark this answers is not in the document]
Comment (rishi-anand) [reply api/k3s_config.go:3]: Sure

**Review by jzhoucliqr — APPROVED** [review]

**Review by paynejacob — APPROVED** [review]

**Pull request merged**